internal/venice-client-common/src/main/java/com/linkedin/venice/utils/AvroSupersetSchemaUtils.java

```diff
@@ -107,32 +107,50 @@ private static Schema unionSchema(Schema s1, Schema s2) {
     return Schema.createUnion(combinedSchema);
   }
 
+  private static void copyFieldProperties(FieldBuilder fieldBuilder, Schema.Field field) {
+    AvroCompatibilityHelper.getAllPropNames(field).forEach(k -> {
+      String propValue = AvroCompatibilityHelper.getFieldPropAsJsonString(field, k);
+      if (propValue != null) {
+        fieldBuilder.addProp(k, propValue);
+      }
+    });
+  }
+
+  private static FieldBuilder deepCopySchemaField(Schema.Field field) {
+    FieldBuilder fieldBuilder = AvroCompatibilityHelper.newField(null)
+        .setName(field.name())
+        .setSchema(field.schema())
+        .setDoc(field.doc())
+        .setOrder(field.order());
+    copyFieldProperties(fieldBuilder, field);
+
+    // set default as AvroCompatibilityHelper builder might drop defaults if there is type mismatch
```

Comment:
Could you explain a bit about this comment? 
If type mismatch, then the field schema is invalid right? Shouldn't we throw exception instead?

Reply:
Actually, I am not sure about this comment and I copied part of the code/comment from the existing function.
I guess it might be fine since f1 default may not be good as the default value for the merged schema.